deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sDeployerUtil.java

```diff
@@ -187,5 +189,64 @@
         logger.debug("Found services {} for app {}", services, context.getAppName());
         return services;
     }
+    
+    /**
+     * Gets app and service name from labels generated by hyscale
+     * For pods not deployed with hyscale AppMetadata with only namespace will be returned
+     * @param podList
+     * @return List of {@link AppMetadata} containing details of deployed apps
+     */
+    public static List<AppMetadata> getAppMetadata(List<V1Pod> podList){
+        if (podList == null) {
+            return null;
+        }
+        Map<String, Map<String, Set<String>>> namespaceVsAppsVsServices = getNamespaceVsAppsVsServices(podList);
+        
+        List<AppMetadata> appsMetadata = new ArrayList<AppMetadata>();
+        
+        namespaceVsAppsVsServices.entrySet().stream().forEach(namespaceVsApps -> {
+            String namespace = namespaceVsApps.getKey();
+            
+            if (namespaceVsApps.getValue() == null || namespaceVsApps.getValue().isEmpty()) {
+                AppMetadata appMetadata = new AppMetadata();
+                appMetadata.setNamespace(namespace);
+                appsMetadata.add(appMetadata);
+                return;
+            }
+            namespaceVsApps.getValue().entrySet().forEach( appsVsServices ->{
+                AppMetadata appMetadata = new AppMetadata();
+                appMetadata.setNamespace(namespace);
+                appMetadata.setAppName(appsVsServices.getKey());
+                appMetadata.setServices(new ArrayList<String>(appsVsServices.getValue()));
+                appsMetadata.add(appMetadata);
+            });
+        });
+        return appsMetadata;
+    }
 
+    private static Map<String, Map<String, Set<String>>> getNamespaceVsAppsVsServices(List<V1Pod> pods) {
+        Map<String, Map<String, Set<String>>> mapping = new HashMap<String, Map<String,Set<String>>>();
```

Comment:
Will namespace->AppMetada not be sufficient ?
Map<String,Map> isn't too complicated ?